mindsdb/integrations/handlers/sharepoint_handler/sharepoint_api.py

```diff
@@ -0,0 +1,429 @@
+import requests
+from datetime import datetime, timezone
+from typing import Text, List, Dict, Any
+import json
+from requests import Response
+
+
+class SharepointAPI:
```

Comment:
Let's document each method please. Can we re use some existing SDK ?

Reply:
Currently Microsoft graph SDK for python is in preview: https://github.com/microsoftgraph/msgraph-sdk-python
They have recommended to not use this in production environment

Reply:
tried to resolve in d4be410003d92952b9893e82ab59261bb6499a4a